src/main/java/duke/Reader.java

```diff
@@ -0,0 +1,97 @@
+package duke;
+
+import java.io.File;
+import java.util.Scanner;
+
+public class Reader {
+    private static final String PATH = "./data/";
+    private static final String FILE = "duke.txt";
+    public static final String EVENT = "E";
+    public static final String DEADLINE = "D";
+    public static final String TODO = "T";
+    private File file;
+    private boolean isFileExists;
+
+    public Reader() {
+        file = new File(PATH + "/" + FILE);
+        isFileExists = file.exists();
+    }
+
+    public boolean isFileExists() {
+        return isFileExists;
+    }
+
+    public TaskManager readFile(TaskManager taskManager) {
+        try {
+            Scanner scanner = new Scanner(file);
+            taskManager = readByLine(taskManager, scanner);
+        } catch (Exception e) { //Future task: exception
+            Ui.printError(e);
+        }
+        return taskManager;
+    }
+
+    private TaskManager readByLine(TaskManager taskManager, Scanner sc) {
+        String line;
+        while (sc.hasNext()) {
+            try {
+                line = sc.nextLine();
+                taskManager.addTask(createTask(line));
+            } catch (Exception e) { //Future task: exception
+                Ui.printError(e);
+            }
+        }
+        return taskManager;
+    }
+
+    private Task createTask(String line) throws DukeException {
+        try {
+            Task task;
+            String[] splitLine = splitStringBySlash(line);
+            switch (getTaskType(splitLine)) {
+            case EVENT:
+                task = new Event(getTaskDescription(splitLine), getTaskDate(splitLine));
```

Comment:
Can consider initialising a variable for getTaskType and getTaskDescription to increase code comprehensibility